Added support for thin provisioned volumes

remixtj opened the pull request:
##### SUMMARY
With the boolean option thin, a volume can be provisioned without space
reservation (thin-provisioning).


##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
na_cdot_volume
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.4.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/remixtj/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

```

remixtj committed:
Added support for thin provisioned volumes

With the boolean option thin, a volume can be provisioned without space
reservation (thin-provisioning).

hulquest commented:
LGTM shipit

amit0701 reviewed (changes requested):
It works for changing the volume to thin provisioning..but the doesnt work for vice versa operations 

E.g. 
thin: False
should use space-reserve: volume

remixtj commented:
@amit0701 space-reserve: volume is the default at creation time, that's why i wasn't setting any value.

Do you mean instead when updating the volume settings?

remixtj committed:
Support for changing current reservation state

Instead of only supporting the option on volume creation time,
we can now use thin option and change guarantee already existing volumes.

remixtj commented:
@amit0701 Now you should be able to change the volume setting from thin to thick and vice-versa when you want, instead of only setting at creation time.

remixtj commented:
@hulquest would you mind to review again? there has been significant changes

maxamillion commented:
friendly re-ping @hulquest

remixtj commented:
ping @hulquest
